src/hooks/useNotifications.ts

```diff
@@ -21,11 +20,11 @@ export const useNotifications = (): void => {
     const initializeNotifications = async (): Promise<void> => {
       if (!client) return
 
-      const permission = await requestAndGetNotificationPermission()
+      const permission = await requestNotifications()
 
-      if (permission === messaging.AuthorizationStatus.AUTHORIZED) {
+      if (permission.granted) {
         setAreNotificationsEnabled(true)
-      } else if (permission === messaging.AuthorizationStatus.DENIED) {
+      } else if (!permission.granted && !permission.canRequest) {
```

Comment:
nit `!permission.granted` is always true here as you are in the `else` part of `if(permission.granted)`

Reply:
Agree (and eslint told me this too 😄). But I find it more clear to have explicitly a case "granted" and a case "not granted and not requestable"